### PR TITLE
Enforce pod security on workloads

### PR DESCRIPTION
**Pull request by danail-branekov**
## Is there a related GitHub Issue?
https://github.com/cloudfoundry/korifi/issues/1220

## What is this change about?
* Set `restricted` pod-security labels on workloads namespace so that pods with containers that do not match the `restricted` policies are rejected
* Introduce a temporary mutating webhook to inject proper security context to kpack build pods to workaround https://github.com/pivotal/kpack/issues/972

## Does this PR introduce a breaking change?
No

## Acceptance Steps
See story

## Tag your pair, your PM, and/or team
@kieron-dev

### CODE REVIEW

File: kpack-image-builder/config/webhook/object_selector_patch.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: admissionregistration.k8s.io/v1
+kind: MutatingWebhookConfiguration
+metadata:
+  name: mutating-webhook-configuration
+webhooks:
+- name: mkpackbuildpod.korifi.cloudfoundry.org
+  objectSelector:
+    matchExpressions:
+      - key: kpack.io/build
```
Comment (tcdowney) [inline kpack-image-builder/config/webhook/object_selector_patch.yaml:9]: Can we make this apply only to pods that have some specific labels we set?
https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#matching-requests-objectselector

There are clusters that use kpack more broadly than just for Korifi use cases and this feels too broad.

**Comment by tcdowney** [comment]
What is the urgency for this? Do we know if the kpack folks would be willing to accept a contribution for https://github.com/pivotal/kpack/issues/972 to get that support in sooner?

File: kpack-image-builder/config/webhook/object_selector_patch.yaml (hunk 1 above)
Comment (kieron-dev) [reply kpack-image-builder/config/webhook/object_selector_patch.yaml:9]: We've added an additional selector on `korifi.cloudfoundry.org/build-workload-name` which means we only make kpack build pods created by korifi.

**Review by kieron-dev — APPROVED** [review]
LGTM

**Pull request merged**